src/main/java/org/marketdesignresearch/mechlib/winnerdetermination/BidBasedWinnerDetermination.java

```diff
@@ -5,24 +5,42 @@
 import com.google.common.collect.ImmutableSet;
 import com.google.common.math.DoubleMath;
 import edu.harvard.econcs.jopt.solver.ISolution;
+import edu.harvard.econcs.jopt.solver.mip.MIP;
 import edu.harvard.econcs.jopt.solver.mip.Variable;
 import org.marketdesignresearch.mechlib.core.*;
+import org.marketdesignresearch.mechlib.core.bid.Bid;
 import org.marketdesignresearch.mechlib.core.bid.Bids;
 import org.marketdesignresearch.mechlib.core.bidder.Bidder;
 import org.marketdesignresearch.mechlib.instrumentation.MipInstrumentation;
 import org.marketdesignresearch.mechlib.metainfo.MetaInfo;
 
 import java.math.BigDecimal;
+import java.math.RoundingMode;
 import java.util.*;
+import java.util.stream.Collector;
 
 public abstract class BidBasedWinnerDetermination extends WinnerDetermination {
 
     private Bids bids;
     // TODO: Make sure we're not running in the same issue as back with SATS with this HashMap
     protected Map<BundleBid, Variable> bidVariables = new HashMap<>();
+    
+    private BigDecimal scalingFactor = new BigDecimal(1);
 
     public BidBasedWinnerDetermination(Bids bids) {
         this.bids = bids;
+        
+        
+        BigDecimal maxValue = bids.getBids().stream().map(Bid::getBundleBids).flatMap(Set::stream).map(BundleBid::getAmount).reduce(BigDecimal::max).get();
+        BigDecimal maxMipValue = new BigDecimal(MIP.MAX_VALUE).multiply(new BigDecimal(.9));
+        
+        if (maxValue.compareTo(maxMipValue) == 1) {
+            this.scalingFactor = maxMipValue.divide(maxValue,RoundingMode.HALF_UP);
+        }
+    }
+    
+    protected BigDecimal getScaledBundleBidAmount(BundleBid bundleBid) {
+    	return bundleBid.getAmount().multiply(this.scalingFactor);
```

Comment:
I think that's fine this way, as long as in the function below called `adaptMIPReslut(..)`, you're referring to the original bids (otherwise the Allocation object will be based on scaled values). Maybe, if you could just add a quick comment in the mentioned function that "this is fine" even when scaled, such that future developers don't panic, that would be great. Thanks!